src/main/java/org/cryptomator/ui/common/Animations.java

```diff
@@ -33,4 +44,31 @@ public void setValue(Double value) {
 		);
 	}
 
+	public static <T> Subscription spinOnCondition(Node node, ObservableValue<T> observableValue, Predicate<T> spinCondition) {
+		var spinAnimation = Animations.createDiscrete360Rotation();
+		spinAnimation.setNode(node);
+		return EasyBind.subscribe(observableValue, val -> {
+			if (spinCondition.test(val)) {
+				spinAnimation.playFromStart();
+			} else {
+				spinAnimation.stop();
+				node.setRotate(0);
+			}
+		});
+	}
```

Comment:
Nice idea. Can we have this even more generic? Sth like `animateConditionally(Animation, Node, Binding<Boolean>)`

ObservableValue + Predicate can be combined to a Binding by the caller.

Reply:
Good Idea. Fixed in 9c9c9f4

Reply:
Please take a look at the first two params as well. :point_up_2: